gax-grpc/src/test/java/com/google/api/gax/grpc/GrpcOperationCallableImplTest.java

```diff
@@ -109,107 +88,142 @@
-  private OperationsStub operationsStub;
+  private FakeChannel initialChannel;
+  private TransportChannel pollTransportChannel;
   private RecordingScheduler executor;
   private ClientContext initialContext;
-  private OperationCallSettings<Integer, Color, Money, Operation> callSettings;
+  private OperationCallSettings<Integer, Color, Currency> callSettings;
 
   private FakeApiClock clock;
   private OperationTimedPollAlgorithm pollingAlgorithm;
 
   @Before
   public void setUp() throws IOException {
-    initialChannel = mock(ManagedChannel.class);
-    pollChannel = mock(ManagedChannel.class);
-    ChannelProvider operationsChannelProvider = mock(ChannelProvider.class);
-    when(operationsChannelProvider.getChannel()).thenReturn(pollChannel);
+    initialChannel = mock(FakeChannel.class);
+    pollTransportChannel = mock(TransportChannel.class);
+    TransportChannelProvider operationsChannelProvider = mock(TransportChannelProvider.class);
+    when(operationsChannelProvider.getTransportChannel()).thenReturn(pollTransportChannel);
 
     clock = new FakeApiClock(0L);
     executor = RecordingScheduler.create(clock);
-    pollingAlgorithm = new OperationTimedPollAlgorithm(FAST_RETRY_SETTINGS, clock);
-
-    OperationsSettings.Builder settingsBuilder = OperationsSettings.defaultBuilder();
-    settingsBuilder
-        .getOperationSettings()
-        .setRetrySettings(FAST_RETRY_SETTINGS.toBuilder().setMaxAttempts(1).build());
-    OperationsSettings settings =
-        OperationsSettings.defaultBuilder()
-            .setTransportProvider(
-                GrpcTransportProvider.newBuilder()
-                    .setChannelProvider(operationsChannelProvider)
-                    .build())
-            .build();
-    operationsStub = GrpcOperationsStub.create(settings);
+    pollingAlgorithm = OperationTimedPollAlgorithm.create(FAST_RETRY_SETTINGS, clock);
 
-    SimpleCallSettings<Integer, Operation> initialCallSettings =
-        SimpleCallSettings.<Integer, Operation>newBuilder()
+    SimpleCallSettings<Integer, OperationSnapshot> initialCallSettings =
+        SimpleCallSettings.<Integer, OperationSnapshot>newBuilder()
             .setRetrySettings(FAST_RETRY_SETTINGS.toBuilder().setMaxAttempts(1).build())
             .build();
 
     callSettings =
-        OperationCallSettings.<Integer, Color, Money, Operation>newBuilder()
+        OperationCallSettings.<Integer, Color, Currency>newBuilder()
             .setInitialCallSettings(initialCallSettings)
-            .setResponseClass(Color.class)
-            .setMetadataClass(Money.class)
+            .setResponseTransformer(new ResponseTransformer())
+            .setMetadataTransformer(new MetadataTransformer())
             .setPollingAlgorithm(pollingAlgorithm)
             .build();
 
     initialContext = getClientContext(initialChannel, executor);
   }
 
+  private static class ResponseTransformer implements ApiFunction<OperationSnapshot, Color> {
```

Comment:
So are there now dedicated tests for the actual transformers somewhere?

Reply:
Yes, in GrpcOperationTransformersTest (see the main PR)